app/scripts/migrations/152.ts

```diff
@@ -0,0 +1,91 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+import {
+  Caip25EndowmentPermissionName,
+  Caip25CaveatType,
+} from '@metamask/chain-agnostic-permission';
+
+export const version = 152;
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+/**
+ * Add sessionProperties property to CAIP-25 permission caveats
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state
+ * @returns Updated versioned MetaMask extension state
+ */
+export async function migrate(originalVersionedData: VersionedData) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+/**
+ * Add sessionProperties property to CAIP-25 permission caveats
+ *
+ * @param state - The MetaMask state
+ * @returns The updated MetaMask state
+ */
+function transformState(state: VersionedData['data']) {
+  if (!hasProperty(state, 'PermissionController')) {
+    return state;
+  }
+
+  if (!isObject(state.PermissionController)) {
+    global.sentry?.captureException?.(
+      new Error(
+        `Migration ${version}: typeof state.PermissionController is ${typeof state.PermissionController}`,
+      ),
+    );
+    return state;
+  }
+
+  const { subjects } = state.PermissionController;
+
+  if (!isObject(subjects)) {
+    global.sentry?.captureException?.(
+      new Error(
+        `Migration ${version}: typeof state.PermissionController.subjects is ${typeof subjects}`,
+      ),
+    );
+    return state;
+  }
+
+  for (const subject of Object.values(subjects)) {
+    if (
+      !isObject(subject) ||
+      !hasProperty(subject, 'permissions') ||
+      !isObject(subject.permissions)
+    ) {
+      continue;
+    }
+
+    const { permissions } = subject;
+    const caip25Permission = permissions[Caip25EndowmentPermissionName];
+
+    if (
+      !isObject(caip25Permission) ||
+      !Array.isArray(caip25Permission.caveats)
+    ) {
+      continue;
+    }
+
+    const caip25Caveat = caip25Permission.caveats.find(
+      (caveat) => caveat.type === Caip25CaveatType,
+    );
+
+    if (
+      caip25Caveat &&
+      isObject(caip25Caveat.value) &&
+      !hasProperty(caip25Caveat.value, 'sessionProperties')
+    ) {
+      caip25Caveat.value.sessionProperties = {};
```

Comment:
What's the reasoning for requiring this property now?

Reply:
It really should always have been required. But what prompted us to add it now is the addition of the Solana AccountChanged event (for our Solana wallet-standard implementation). Callers [can request to be notified when the selected Solana account in the wallet changes](https://github.com/MetaMask/metamask-extension/pull/31534) by including the `solana_accountChanged_notifications: true` entry in the sessionProperties

Reply:
We just plain weren't storing it previously since we weren't enforcing any of the properties (e.g. `expiry`) in there. We've added a `solana_accountChanged_notifications` sessionProperty intended for our wallet-standard library to identify itself from other Multichain API connections so that the wallet knows that it should send solana accountChange events to the dapp